src/tests/GfxBatchHbaoTest.cpp

```diff
@@ -0,0 +1,317 @@
+#include <Corrade/Containers/Optional.h>
+#include <Corrade/PluginManager/Manager.h>
+#include <Corrade/Utility/Path.h>
+#include <Magnum/DebugTools/CompareImage.h>
+#include <Magnum/GL/Context.h>
+#include <Magnum/GL/Extensions.h>
+#include <Magnum/GL/Framebuffer.h>
+#include <Magnum/GL/Mesh.h>
+#include <Magnum/GL/OpenGLTester.h>
+#include <Magnum/GL/Renderer.h>
+#include <Magnum/GL/Texture.h>
+#include <Magnum/GL/TextureFormat.h>
+#include <Magnum/GL/Version.h>
+#include <Magnum/Image.h>
+#include <Magnum/ImageView.h>
+#include <Magnum/Math/Color.h>
+#include <Magnum/Math/Matrix4.h>
+#include <Magnum/MeshTools/Compile.h>
+#include <Magnum/PixelFormat.h>
+#include <Magnum/Shaders/PhongGL.h>
+#include <Magnum/Trade/AbstractImporter.h>
+#include <Magnum/Trade/ImageData.h>
+#include <Magnum/Trade/MeshData.h>
+
+#include "esp/gfx_batch/Hbao.h"
+
+#include "configure.h"
+
+namespace {
+
+namespace Cr = Corrade;
+namespace Mn = Magnum;
+using namespace Mn::Math::Literals;
+
+struct GfxBatchHbaoTest : Mn::GL::OpenGLTester {
+  explicit GfxBatchHbaoTest();
+  void generateTestData();
+  void test();
+
+  Mn::Matrix4 projMatrices[2] = {
+      Mn::Matrix4::perspectiveProjection(45.0_degf, 1.0f, 0.1f, 100.0f),
+      Mn::Matrix4::orthographicProjection(Magnum::Vector2(4.0f, 4.0f),
+                                          0.1f,
+                                          100.0f)};
+  std::string srcRGBFileNames[2] = {"van-gogh-room.color.png",
+                                    "van-gogh-room.color-ortho.png"};
+
+  std::string srcDepthFileNames[2] = {"van-gogh-room.depth.exr",
```

Comment:
`std::string` :scream: 

Also, these could all be a file-local struct, and the instanced data could then reference them:

```cpp
struct Projection {
    Mn::Matrix4 projection;
    const char* sourceColorFilename;
    const char* sourceDepthFilename;
};

const Projection PerspectiveData{
    ...
};

const Projection OrthographicData{
    ...
};
```

...

Reply:
Yeah, thanks, that's much better